ADDED $sortByPostion flag to getChildren()

denisristic opened the pull request:
Method "getChildren" sort ordering (SQUASHTOBERFEST)

### Description
ADDED $sortByPostion flag to getChildren()

### Fixed Issues (if relevant)
1. magento/magento2#11310

Commit:
ADDED $sortByPostion flag to getChildren()

ishakhsuvarov commented:
@denisristic Please consider covering the updated behavior with integration test.

Commit:
ADDED $sortByPosition flag to Magento\Catalog\Model\Category
ADDED test testGetChildrenSroted() to integration testsuite Magento\Catalog\Model\Category

Commit:
FIXED integration test
ADDED $recursive and $isActive, flags to getChildren()

Commit:
FIXED sort

Commit:
FIXED default getChildren() $recursive value
